rpcs3/Emu/Io/usio.cpp

```diff
@@ -556,56 +513,38 @@ void usb_device_usio::interrupt_transfer(u32 buf_size, u8* buf, u32 endpoint, Us
 		{
 			usio_log.trace("UsioWrite(Channel: 0x%02X, Register: 0x%04X, Length: 0x%04X)", usio_channel, usio_register, usio_length);
 			ensure(((~(usio_register >> 8)) & 0xF0) == buf[1]);
-
 			expecting_data = true;
 			usio_data.clear();
-			return;
 		}
-
-		if ((buf[0] & USIO_COMMAND_READ) == USIO_COMMAND_READ)
+		else if ((buf[0] & USIO_COMMAND_READ) == USIO_COMMAND_READ)
 		{
 			usio_log.trace("UsioRead(Channel: 0x%02X, Register: 0x%04X, Length: 0x%04X)", usio_channel, usio_register, usio_length);
+			response_seek = 0;
+			response.clear();
 			usio_read(usio_channel, usio_register, usio_length);
-			return;
 		}
-
-		// Init and reset commands
-		if ((buf[0] & 0xA0) == 0xA0)
+		else if ((buf[0] & USIO_COMMAND_INIT) == USIO_COMMAND_INIT) // Init and reset commands
 		{
-			const std::array<u8, 2> init_command = {0xA0, 0xF0}; // This kind of command starts with 0xA0, 0xF0 commonly. For example, {0xA0, 0xF0, 0x28, 0x00, 0x00, 0x80}
-			ensure(memcmp(buf, init_command.data(), 2) == 0);
-			return;
+			//const std::array<u8, 2> init_command = {0xA0, 0xF0}; // This kind of command starts with 0xA0, 0xF0 commonly. For example, {0xA0, 0xF0, 0x28, 0x00, 0x00, 0x80}
+			//ensure(memcmp(buf, init_command.data(), 2) == 0);
```

Comment:
why is this commented out and why is there no comment for it?

Reply:
They literally do nothing and I don't think it is necessary to do the assertion for `USIO_COMMAND_INIT`.
It can't lead to problems if `memcmp(buf, init_command.data(), 2) != 0`.